rustls/tests/unbuffered.rs

```diff
@@ -15,17 +15,106 @@ mod common;
 
 const MAX_ITERATIONS: usize = 100;
 
+const CLIENT_TRANSCRIPT_TLS13: &[&str] = &[
+    "EncodedTlsData",
```

Comment:
Really, this should be in terms of the public API (`ConnectionState`) rather than the test code (`State`). As it is, this tests the test code more than the library. This also needs to extend to every test in this file.

I would also separate the tests by version (eg, `fn tls12_handshake`, `fn tls13_handshake`), and put the expectations inline in the test rather than at the top level.

Reply:
For the sake of clarity and speed I've push a couple of commits to https://github.com/rustls/rustls/tree/jbp-unbuffered-api-testing which do this -- feel free to review and squash those into your top commit?

Reply:
thanks, I just squashed those into my last commit